scripts/ObstacleDetector.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+import rospy
+import math
+from std_msgs.msg import Bool, Header, String, Float64, Float32
+from sensor_msgs.msg import LaserScan
+import genpy
+import numpy as np
+from geometry_msgs.msg import Point,PointStamped
+class ObstacleDetector():
+    def __init__(self):
+        self.pub0 = rospy.Publisher('/estop', Bool,  queue_size=10) # init your publishers early
+        self.pubObstPoint = rospy.Publisher('/point2follow', PointStamped, queue_size=10)
+        rospy.init_node('ObstacleDetector', anonymous=True)
+        self.subScan = rospy.Subscriber('/front/scan',LaserScan,self.callback)
+        self.rate = rospy.Rate(1)
+        self.scan = None
+
+    def callback(self,data):
+        self.scan = data
+
+    def detectObstacles(self):
```

Comment:
So . . . function docstrings and multiline comments exist for this . . . Not sure it's important now, but for code cleanliness, this could be much better.